### PR TITLE
Check if database proposal exists in CrowbarService::dump_openstack_database

### PR DESCRIPTION
**Pull request by mjura**
Check if database proposal exists in case,
when somebody is trying to upgrade undeployed nodes
or only with Ceph cluster

### CODE REVIEW

**Comment by jsuchome** [comment]
Wouldn't it be better to do the check for proposal at the beginning and skip the search for db nodes?

**Comment by mjura** [comment]
> Wouldn't it be better to do the check for proposal at the beginning and skip the search for db nodes?

Unfortunately, no because we skip 
```
      node["crowbar_wall"]["crowbar_upgrade_step"] = step
      node.save
```
which is needed to move forward upgrade process.

**Comment by jsuchome** [comment]
Hm, OK, I think it could stay this way.

File: crowbar_framework/app/models/crowbar_service.rb [hunk 1]
```diff
@@ -103,7 +103,10 @@ def dump_openstack_database
       node.save
     end
 
-    raise "There does not seem to exist any node running the database." if db_nodes.empty?
+    # Check if database proposal exists in case when somebody is trying to upgrade undeployed nodes
+    # or only with Ceph cluster
+    db_proposal = Proposal.where(barclamp: "database", name: "default").first
```
Comment (vuntz) [inline crowbar_framework/app/models/crowbar_service.rb]: Can you move that line outside of the loop? Also, please don't look for a proposal with name `default` as the name could be different; just look for any database proposal.

**Comment by mjura** [comment]
Was tested, works fine. Please comment

File: crowbar_framework/app/views/installer/upgrades/backup.html.haml [hunk 2]
```diff
@@ -17,8 +17,11 @@
               = t(".description")
 
             %p
-              = database_node.name
-              = database_backup_path
+              - if upgrade_database_node
+                = upgrade_database_node.name
+                = upgrade_database_backup
+              - else
+                No database to dump
```
Comment (jsuchome) [inline crowbar_framework/app/views/installer/upgrades/backup.html.haml]: Could you make full sentence out of this?

**Comment by vuntz** [comment]
@mjura can you rebase?

**Comment by mjura** [comment]
@vuntz : It was rebased

**Comment by mjura** [comment]
Ready to go

**Comment by jsuchome** [comment]
+1c (the texts should be translated, but I'm not sure if we need to go through them now...)

**Comment by vuntz** [comment]
If the text is not made localizable in this PR, then please create a followup task somewhere to make sure it will be.

**Comment by mjura** [comment]
I've added translation, but I've to test

File: crowbar_framework/app/models/crowbar_service.rb [hunk 3]
```diff
@@ -103,7 +103,11 @@ def dump_openstack_database
       node.save
     end
 
-    raise "There does not seem to exist any node running the database." if db_nodes.empty?
+    # Check if database proposal exists in case when somebody is
+    # trying to upgrade undeployed nodes or only with Ceph cluster
+    db_proposal = Proposal.where(barclamp: "database").first
```
Comment (MaximilianMeister) [inline crowbar_framework/app/models/crowbar_service.rb]: MINOR: you could also use 
`db_proposal = Proposal.find_by(barclamp: "database")` 
or
`db_proposal = Proposal.find_by(barclamp: "database", name: "default")`

**Comment by rsalevsky** [comment]
@mjura Can you integrate #308 and close my PR?

File: crowbar_framework/app/models/crowbar_service.rb (hunk 3 above)
Comment (vuntz) [inline crowbar_framework/app/models/crowbar_service.rb]: (don't do the second suggestion, first one is fine, I guess)

Comment (mjura) [inline crowbar_framework/app/models/crowbar_service.rb]: Ok, I've added first one

**Comment by mjura** [comment]
@rsalevsky : I've cherry-picked your changed and rebased it

**Comment by vuntz** [comment]
+1

**Comment by rsalevsky** [comment]
+1

**Pull request merged**